pages/en/zkapps/how-to-write-a-zkapp.mdx

```diff
@@ -474,6 +474,40 @@ super.deploy(args);
 If you don't need to add anything else to `deploy`, there's no need to override it, just leave it out.
 In the example above however, we are setting our state `x` to `Field(10)`. Uninitialized values will be zero.
 
+##### Custom data types
+
+As mentioned previously, smart contract `@method` arguments can be any of the built-in SnarkyJS types.
+
+However, what if you want to define your own data types?
+
+You can create your own custom data type for your smart contract using the `CircuitValue` class that SnarkyJS exposes. To start using the `CircuitValue` class to define your data types, you start by extending the `CircuitValue` class.
```

Comment:
Would suggest rephrasing this one sentence as the below. I think it'll fit in well with the rest of your text and read smoothly.
> To start using the `CircuitValue` class to define your data types, you start by extending the `CircuitValue` class.

To do this, create a class that extends `CircuitValue`.

Reply:
I like the change, fixed!